### PR TITLE
Add and improve the function documentations

### PR DESCRIPTION
**Pull request by beam2d**
This PR includes following changes:
- I fixed some wording of NegativeSampling.
- I wrote a documentation of InceptionBN.
- I added the InceptionBN, matmul, and batch_matmul to the reference.

### CODE REVIEW

**Comment by delta2323** [comment]
I'll check it.

File: chainer/functions/negative_sampling.py [hunk 1]
```diff
@@ -9,24 +9,24 @@
...
 
     .. math::
 
        f(x, p) = \log\sigma(x^\\top w_p) + \\
        k E_{i \sim P(i)}[\log\sigma(- x^\\top w_i)],
 
-    where :math:`\sigma(\cdot)` is a sigmoid function, :math:`w_i` is a wegith
-    vector for a word :math:`i`, and :math:`p` is a positive example.
-    It is approximeted with :math:`k` examples :math:`N` sampled from probabity
+    where :math:`\sigma(\cdot)` is a sigmoid function, :math:`w_i` is the weight
```
Comment (delta2323) [inline chainer/functions/negative_sampling.py]: This line exceeds 79 characters.

File: chainer/functions/negative_sampling.py [hunk 2]
```diff
@@ -9,24 +9,24 @@
...
+    vector for the word :math:`i`, and :math:`p` is a positive example.
+    It is approximeted with :math:`k` examples :math:`N` sampled from probability
```
Comment (delta2323) [inline chainer/functions/negative_sampling.py]: This line exceeds 79 characters.

**Comment by delta2323** [comment]
LGTM except two commented points.

**Comment by beam2d** [comment]
I fixed the lines.

**Comment by delta2323** [comment]
Merged.

**Pull request merged**